### PR TITLE
Feat/content extension messaging

### PR DESCRIPTION
**Pull request by sabjorn**
fixes issue with previous storage mechanism by introducing a background script which manages data through an `indexeddb` database.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,16 +4,20 @@
   "description": "Chrome extension adding extra features to the Bandcamp experience.",
   "main": "index.js",
   "dependencies": {
+    "webpack": "^4.43.0"
```
Comment (emcniece) [inline package.json]: Very minor note: I think Webpack could be considered a dev dependency, the [install docs](https://webpack.js.org/guides/installation/) note it as such. There is no real impact in this project, but as a general rule the packages listed in the `dependencies: {}` are critical to the runtime of the application, while dev- and build-time packages get listed in `devDependencies`.

If you decide to adjust this (again, not high priority) you could cut-paste the webpack line into `devDependencies`.

File: package.json [hunk 2]
```diff
@@ -4,16 +4,20 @@
...
+    "eslint-plugin-prettier": "^3.1.2",
+    "idb": "^5.0.3",
+    "prettier": "^1.19.1",
+    "webpack-cli": "^3.3.11"
   },
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "lint": "npx eslint ./src/"
+    "lint": "npx eslint ./src/",
+    "build": "webpack --config webpack.config.js"
```
Comment (emcniece) [inline package.json]: Since you're using `npx` in the `lint` script, you might consider doing the same here like `npx webpack`. The current command relies on Webpack being globally installed by NPM, and since you list Webpack up in `[dev]Dependencies` it is being installed locally.

It works fine because of prior env setup (you very likely installed webpack globally at some point), so no need to change if you want, but you may find that fresh envs and installs may get stuck here because the README setup/install steps do not list `npm install -g webpack`. 

Tl;dr: consider one of the following:
- Adjust this line to `npx webpack ...`
- Include a global webpack install step in the readme

File: src/background.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
+console.log = function() {}; // disable logging
```
Comment (emcniece) [inline src/background.js:1]: I know this existed earlier, but it's here because of a file shift, so I'll comment on it again 😁 

This is dangerous, applies to the entire JS VM, and may screw up other libraries around you! Please consider using a custom local logger in the future, there's lots of options and it would be easy to roll your own too.

Using a custom logger could enable the creation of a "debug" checkbox in `browser_action.html` in the future to enable/disable verbose logging.

File: src/download_helper.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        url = $(item).attr('href');
+        filelist += "curl -OJ " + url + " \\ &\n";
+    })
+    return filelist
+};
+
+function download(filename, text) {
+  var element = document.createElement('a');
+
+  element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
```
Comment (emcniece) [inline src/download_helper.js:13]: Consistency nit: `generateDownloadList()` uses jQuery for DOM traversal, while `download()` uses raw JS. The `download()` method could use jQuery to create the anchor element a bit more cleanly, or the generate method could use native JS to obtain `filelist`. Inconsequential either way, so long as jQuery is globally available.

File: src/label_view.js [hunk 5]
```diff
@@ -1,229 +1,171 @@
...
-      // load storage
-      if (isEmpty(result)) {
-        console.log("storage empty, storing new");
-        chrome.storage.sync.set({ previews: storageCache });
-      } else {
-        console.log("storage exists, storing to variable 'storageCache'");
-        storageCache = result["previews"];
-      }
+let previewId; // globally stores which 'preview' button was last clicked
+let previewOpen = false; // globally stores if preveiw window is open
```
Comment (emcniece) [inline src/label_view.js:6]: These globals might be good candidates for storing on `pluginState` in the future 👍 

I'm not 100% sure about using `let` block-level scoping in the global JS VM, but I have a feeling that these vars will truly be global and thus vulnerable to external manipulation or naming collisions.

File: package.json [hunk 6]
```diff
@@ -2,18 +2,25 @@
...
+    "prettier": "^1.19.1",
+    "webpack-cli": "^3.3.11"
   },
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "lint": "npx eslint ./src/"
+    "lint": "npx eslint ./src/",
+    "build": "webpack --config webpack.config.js",
+    "clean": "rm -fr ./node_modules ./dist",
+    "package": "npm run build && zip BandcampLabelView.zip LICENSE _locales/**/* icons/* dist/* manifest.json html/browser_action.html"
```
Comment (emcniece) [inline package.json:23]: If this line gets out of hand, it's not uncommon to make a `package.sh` script.
